src/__tests__/DependencyGraph-test.js

```diff
@@ -90,14 +90,14 @@ describe('DependencyGraph', function() {
       cache: new Cache(),
       fileWatcher,
       providesModuleNodeModules: [
-        'haste-fbjs',
-        'react-haste',
-        'react-native',
+        { name: 'haste-fbjs', parent: null },
+        { name: 'react-haste', parent: null },
+        { name: 'react-native', parent: null },
```

Comment:
We need support for this use case:

```
- node_modules
  - react-native
    - node_modules
      - fbjs
  - react
    - node_modules
      - fbjs
```

If we're not checking for the `providesModuleNodeModule`'s parent module (as we weren't before), this will break. What this PR does is allow us to specify that we absolutely want to use the fbjs that's a dep of react-native...not the fbjs that's a peer of react-native (in the case of npm3 and Relay) or fbjs that is a dep of React.

As you know https://github.com/facebook/react-native/pull/5084 makes the fbjs use-case moot...but that's not merged yet, and perhaps there could be other cases in the future where this is necessary.

Anyway, as I explained in my PR description, it seems that the current behavior is the _wrong_ behavior, as it's not at all deterministic, which is exactly the problem (or one of the problems haha) that Haste aims to solve.